src/main/java/com/google/devtools/build/lib/remote/blobstore/CombinedDiskHttpBlobStore.java

```diff
@@ -110,40 +117,30 @@ public boolean containsKey(String key) {
   @Override
   public boolean getActionResult(String key, OutputStream out)
       throws IOException, InterruptedException {
-    if (super.getActionResult(key, out)) {
-      return true;
-    }
-
-    try (ByteArrayOutputStream tmpOut = new ByteArrayOutputStream()) {
-      if (bsHttp.getActionResult(key, tmpOut)) {
-        byte[] tmp = tmpOut.toByteArray();
-        super.putActionResult(key, tmp);
-        ByteStreams.copy(new ByteArrayInputStream(tmp), out);
-        return true;
-      }
-    }
-
-    return false;
+    return getFromFuture(get(key, out));
```

Comment:
Note, `OnDiskBlobStore` [applies a prefix to the key](https://github.com/bazelbuild/bazel/blob/fbeb0c9793e146d08c9d6dfdfceaea8b24f86e6c/src/main/java/com/google/devtools/build/lib/remote/blobstore/OnDiskBlobStore.java#L59-L63). So, this implementation would miss action cache items written by `OnDiskBlobStore`. That's the reason why the combined cache calls `super.getActionResult` and `bsHttp.getActionResult` separately.

Reply:
Right, I missed the prefix in this case... Thanks!